compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -2083,18 +2083,11 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
         output: Option<&'tcx hir::Ty<'tcx>>,
     ) {
         debug!("visit_fn_like_elision: enter");
-        let mut arg_elide = Elide::FreshLateAnon(Cell::new(0));
-        let arg_scope = Scope::Elision { elide: arg_elide.clone(), s: self.scope };
+        let arg_scope = Scope::Elision { elide: Elide::FreshLateAnon(Cell::new(0)), s: self.scope };
         self.with(arg_scope, |_, this| {
             for input in inputs {
                 this.visit_ty(input);
             }
-            match *this.scope {
-                Scope::Elision { ref elide, .. } => {
-                    arg_elide = elide.clone();
-                }
-                _ => bug!(),
```

Comment:
If this check should stay around for safety reasons, it should probably be converted to an assert. As is, it looks unintentional and just as a byproduct of satisfying the `match` syntax.

Reply:
I agree that this looks like it is left over from the `match` syntax.